lacework/resource_lacework_alert_channel_datadog.go

```diff
@@ -0,0 +1,256 @@
+package lacework
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/lacework/go-sdk/api"
+)
+
+func resourceLaceworkAlertChannelDatadog() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLaceworkAlertChannelDatadogCreate,
+		Read:   resourceLaceworkAlertChannelDatadogRead,
+		Update: resourceLaceworkAlertChannelDatadogUpdate,
+		Delete: resourceLaceworkAlertChannelDatadogDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: importLaceworkIntegration,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"intg_guid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"datadog_site": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(api.DatadogSiteCom),
+				ValidateFunc: func(value interface{}, key string) ([]string, []error) {
+					switch value.(string) {
+					case string(api.DatadogSiteEu), string(api.DatadogSiteCom):
+						return nil, nil
+					default:
+						return nil, []error{
+							fmt.Errorf(
+								"%s: can only be either '%s' or '%s'", key,
+								api.DatadogSiteEu, api.DatadogSiteCom,
+							),
+						}
+					}
+				},
+			},
+			"datadog_service": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(api.DatadogServiceLogsDetails),
+				ValidateFunc: func(value interface{}, key string) ([]string, []error) {
+					switch value.(string) {
+					case string(api.DatadogServiceLogsDetails), string(api.DatadogServiceLogsSummary), string(api.DatadogServiceEventsSummary):
+						return nil, nil
+					default:
+						return nil, []error{
+							fmt.Errorf(
+								"%s: can only be either '%s', '%s' or '%s'", key,
+								api.DatadogServiceLogsDetails, api.DatadogServiceLogsSummary, api.DatadogServiceEventsSummary,
+							),
+						}
+					}
+				},
+			},
+			"api_key": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"created_or_updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_or_updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"org_level": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceLaceworkAlertChannelDatadogCreate(d *schema.ResourceData, meta interface{}) error {
+	site, _ := api.DatadogSite(d.Get("datadog_site").(string))
+	service, _ := api.DatadogService(d.Get("datadog_service").(string))
```

Comment:
Note to self: These errors are safe to ignore since the schema above will error if the user does not provide a valid value. 💯